rtgui/cropwindow.cc

```diff
@@ -272,210 +269,209 @@ void CropWindow::buttonPress (int button, int type, int bstate, int x, int y)
 {
 
     bool needRedraw = true;  // most common case ; not redrawing are exceptions
+    EditSubscriber *editSubscriber = iarea->getCurrSubscriber();
 
     iarea->grabFocus (this);
 
-    if (button == 1 && type == GDK_2BUTTON_PRESS && onArea (CropImage, x, y) && (state == SNormal || state == SCropImgMove)) {
-        if (fitZoomEnabled) {
-            if (fitZoom) {
-                state = SNormal;
-                zoomVersion = exposeVersion;
-                screenCoordToImage (x, y, action_x, action_y);
-                changeZoom (ZOOM11INDEX, true, action_x, action_y);
-                fitZoom = false;
+    if (button == 1) {
```

Comment:
The logic that follows look hard to read and hence brittle to maintain. I am not sure how to go about it exactly, but maybe the repetitive parts could be change to be data-driven instead of logic-driven? For example have a list of possible crop areas and the associated values the various variables should have if the event happened inside that area?